asdf/extension/_legacy.py

```diff
@@ -248,11 +258,35 @@ def resolver(self):
         return self.extension_list.resolver
 
 
-default_extensions = _DefaultExtensions()
+_default_extensions = _DefaultExtensions()
+
+
+def __getattr__(name):
+    if name == "default_extensions":
+        warnings.warn(
+            "default_extensions is deprecated. "
+            "Please see the new extension API "
+            "https://asdf.readthedocs.io/en/stable/asdf/extending/converters.html",
+            AsdfDeprecationWarning,
+        )
+        return _default_extensions
+    msg = f"module {__name__!r} has no attribute {name!r}"
+    raise AttributeError(msg)
```

Comment:
I'm less sure about this one, but maybe this deprecation warning could just get moved into the `__getattr__` that has already been added to `asdf.extension.__init__`?